sdk/core/azure_core_amqp/src/messaging.rs

```diff
@@ -1075,74 +1105,95 @@ impl AmqpMessage {
         self.footer.as_ref()
     }
 
-    pub fn set_header(&mut self, header: AmqpMessageHeader) {
-        self.header = Some(header);
-    }
-
-    pub fn set_properties(&mut self, properties: AmqpMessageProperties) {
-        self.properties = Some(properties);
-    }
-
-    pub fn set_message_annotations(&mut self, message_annotations: impl Into<AmqpAnnotations>) {
-        self.message_annotations = Some(message_annotations.into());
-    }
-
-    pub fn set_message_body(&mut self, body: impl Into<AmqpMessageBody>) {
-        self.body = body.into();
+    /// Set the message ID for the message.
+    ///
+    /// This will overwrite any existing message ID on the message.
+    ///
+    /// # Arguments
+    ///
+    /// * `message_id` - The message ID to set on the message.
+    ///
+    /// # Example
```

Comment:
Nit: even if there's one, idiomatically it's always "# Examples". Several other instances I see in this PR at least.